gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/v4/impl/FlowServiceImpl.java

```diff
@@ -139,19 +142,35 @@ public List<Flow> findByReference(final FlowReferenceType flowReferenceType, fin
     public List<Flow> save(final FlowReferenceType flowReferenceType, final String referenceId, final List<Flow> flows) {
         try {
             log.debug("Save flows for reference {},{}", flowReferenceType, flowReferenceType);
-            flowRepository.deleteByReference(flowReferenceType, referenceId);
-            if (flows == null) {
+            if (flows == null || flows.isEmpty()) {
+                flowRepository.deleteByReference(flowReferenceType, referenceId);
                 return List.of();
-            } else {
-                List<Flow> createdFlows = new ArrayList<>();
-                for (int order = 0; order < flows.size(); ++order) {
-                    io.gravitee.repository.management.model.flow.Flow createdFlow = flowRepository.create(
-                        flowMapper.toRepository(flows.get(order), flowReferenceType, referenceId, order)
-                    );
-                    createdFlows.add(flowMapper.toDefinition(createdFlow));
+            }
+            Map<String, io.gravitee.repository.management.model.flow.Flow> dbFlowsById = flowRepository
+                .findByReference(flowReferenceType, referenceId)
+                .stream()
+                .collect(Collectors.toMap(io.gravitee.repository.management.model.flow.Flow::getId, Function.identity()));
+
+            Set<String> flowIdsToSave = flows.stream().map(Flow::getId).filter(Objects::nonNull).collect(Collectors.toSet());
+
+            for (String dbFlowId : dbFlowsById.keySet()) {
+                if (!flowIdsToSave.contains(dbFlowId)) {
+                    flowRepository.delete(dbFlowId);
+                }
+            }
```

Comment:
Do you think that we should add a deleteAllById method in the repository. This method would take a list of id. It would allow us to avoid to loop on delete queries and do a single delete. 🤷🏼  Wdyt ?

Reply:
good point, I will add this method